htlcswitch/switch.go

```diff
@@ -543,13 +543,38 @@ func (s *Switch) CleanStore(keepPids map[uint64]struct{}) error {
 	return s.Store.CleanStore(keepPids)
 }
 
-// SendHTLC is used by other subsystems which aren't belong to htlc switch
-// package in order to send the htlc update. The attemptID used MUST be unique
-// for this HTLC, and MUST be used only once, otherwise the switch might reject
-// it.
+// SendHTLC attempts to forward an HTLC to the given first hop using the
+// specified attempt ID. It can be used by other subsystems in order to dispatch
+// a payment attempt. The attemptID MUST be globally unique across the lifetime
+// of the payment.
+//
+// Duplicate attemptIDs are rejected to ensure correctness of the
+// asynchronous payment result store and to prevent reuse after completion.
+//
+// The Switch guarantees that only one HTLC will be forwarded for a given
+// attemptID, and will return ErrDuplicateAdd for subsequent uses.
+//
+// This method is safe to call from remote clients via SendOnion or from
+// a local ChannelRouter.
 func (s *Switch) SendHTLC(firstHop lnwire.ShortChannelID, attemptID uint64,
 	htlc *lnwire.UpdateAddHTLC) error {
 
+	// Safety check to ensure that this attempt ID is not currently created
+	// within the result store.
+	err := s.Store.InitAttempt(attemptID)
```

Comment:
I think we'll need to think about if we should undo the storage of the placeholder if we encounter any errors to send the attempt out to the peer.